src/components/attribute-selector.tsx

```diff
@@ -1,39 +1,74 @@
-import React from "react";
+import React, { useState } from "react";
 import { useStateContext } from "../hooks/use-state";
+import ToggleIcon from "../assets/images/icon-toggle.svg";
 
 import "./attribute-selector.scss";
 
+const kHourlyAttributes = ["Dew Point", "Barometric Pressure at sea level", "Air Temperature", "Visibility",
+                            "Wind Direction", "Wind Speed", "Precipitation in last hour"];
+const kMonthlyDailyAttributes = ["Average temperature", "Precipitation", "Max temperature", "Min temperature",
+                                  "Snowfall", "Average wind speed"];
+
 export const AttributesSelector = () => {
   const {state, setState} = useStateContext();
   const {units} = state;
+  const [allSelected, setAllSelected] = useState(false);
 
-  const selected = false;
-  const attributes = ["All", "Average temperature", "Precipitation", "Max temperature", "Min temperature",
-                        "Snowfall", "Average wind speed"];
+  const attributes = state.frequency === "hourly" ? kHourlyAttributes : kMonthlyDailyAttributes;
 
   const handleUnitsClicked = () => {
     setState(draft => {
       draft.units = draft.units === "standard" ? "metric" : "standard";
     });
   };
 
+  const handleSelectAllAttrs = () => {
+    setAllSelected(true);
+    setState(draft => {
+      draft.attributes = state.frequency === "daily" ? kHourlyAttributes : kMonthlyDailyAttributes;
+    });
+  };
+
+  const handleAttributeSelect = (e: React.MouseEvent<HTMLDivElement>) => {
+    const attrSelected = e.currentTarget.textContent;
+    if (allSelected) {
+      setAllSelected(false);
+    }
+    setState(draft => {
+      if (allSelected) {
+        draft.attributes = [];
+      }
+      attrSelected && draft.attributes.push(attrSelected);
```

Comment:
It seems like this doesn't handle toggling the attributes but rather just keeps adding them to the list.

Reply:
Same comment as Doug - it seems we should check if the clicked attribute is already selected, and de-select it if so.